chromium_src/components/content_settings/core/common/cookie_settings_base.cc

```diff
@@ -314,8 +315,14 @@ CookieSettingsBase::GetCurrentCookieSettingWithBraveMetadata() {
         setting_info.secondary_pattern.MatchesAllHosts();                \
   }
 
+// Force GetCookieSetting to use a first-party context when calling into
+// GetCookieSettingInternal, rather than assuming a third-party context. See
+// https://chromium.googlesource.com/chromium/src/+/1eca8080b750b1b3e3067cfb7209163b9026de8a
+#define SiteForCookies() SiteForCookies::FromUrl(first_party_url)
```

Comment:
why would we do this? This seems very concerning.

Reply:
remember that the "why" is the most important thing in comments. I can read the code to see what we're doing, but I don't understand why we would do this

Reply:
It was necessary in order to support ephemeral storage, but I would probably need input from @goodov as to why it's specifically required. The upstream change had this to say:

```
    Rewrite GetCookieSetting to stop assuming a same-site context

    Of the 3 non-test callers, one creates a synthetic first-party
    context, one creates a synthetic third-party context, and one is
    ambiguous. This updates the first-party and ambiguous callers to
    use a new override that accepts a SiteForCookies explicitly; and
    updates the existing method to assume a third-party context, rather
    than a first-party context.
```

Reply:
there should definitely be some explanation in the comment explaining why exactly we're doing this and what impact it has

Reply:
Updated comment to reflect that this is needed for ephemeral storage, but could use @goodov's help as far as fleshing the comment out more fully to cover the "why" d8e24906d7d229863078553577d767a267e95211

Reply:
I will look into this here: https://github.com/brave/brave-browser/issues/39225